src/core/utils/url/cleanUrl.ts

```diff
@@ -5,8 +5,16 @@ export function getSearchParams(path: string) {
 }
 
 /**
- * Removes query parameters and hashes from a given URL string.
+ * Removes search parameters and the fragment
+ * from a given URL string.
  */
 export function cleanUrl(path: string): string {
+  // If the path ends with an optional path parameter,
+  // return it as-is.
+  if (path.endsWith('?')) {
```

Comment:
Instead of analyzing the last segment, I've added a check if the `path` ends with `?`. It will only end with a question mark in the case when it ends with an optional path parameter that must be preserved. 

If the path ends with a search string, it will always have value after `?`, so this check won't pass. 

@KaiSpencer, what do you think about this approach?

Reply:
Awesome! I will wait for the CI and merge this. Then, merge the main feature branch and have this released. Thank you for helping me with this!